codalab/worker/file_util.py

```diff
@@ -268,7 +268,15 @@ def __enter__(self) -> IO[bytes]:
                     fs = TarFileStream(tf, finfo)
                     return GzipStream(fs) if self.gzipped else fs
         else:
-            # Stream a file from disk storage.
+            # Stream a directory or file from disk storage.
+            if os.path.isdir(self.path):
+                if not self.gzipped:
+                    raise IOError("Directories must be gzipped.")
+                return tar_gzip_directory(self.path)
+            if self.gzipped:
+                raise IOError(
```

Comment:
does it make sense to add a comment to the function so future callers know this?

Reply:
done